packages/@o3r/core/schematics/ng-add/updates-for-v8/imports/update-imports-from-v7-to-v8.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-console */
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getSourceFilesFromWorkspaceProjects } from '@o3r/schematics';
+import * as ts from 'typescript';
+import { mapImportV7toV8, renamedPackages } from './v7-to-v8-map-object';
+import { logging } from '@angular-devkit/core';
+import { findNodes } from '@schematics/angular/utility/ast-utils';
+
+/**
+ * Extracted symbol from an import line.
+ */
+interface ExtractedImport {
+  /** Element imported */
+  symbol: string;
+
+  /** Module from where the symbol is imported */
+  location: string;
+}
+
+
+function updateImportsInFile(
+  logger: logging.LoggerApi,
+  tree: Tree,
+  sourceFile: ts.SourceFile,
+  importsRegexp: RegExp,
+  renamePackagesRegexp: RegExp) {
+  const oldImportedSymbolsPerPackage: Record<string, ExtractedImport[]> = {};
+  const unResolvedImports: ExtractedImport[] = [];
+  const importNodes: ts.ImportDeclaration[] = [];
+
+  // First we look for all imports lines targeting an Otter package for which we know a mapping
+  findNodes(sourceFile, ts.SyntaxKind.ImportDeclaration).map((nodeImp) => {
+    const imp = nodeImp as ts.ImportDeclaration;
+    const importFrom = imp.moduleSpecifier.getText().replace(/['"]/g, '');
+
+    const renamePackageMatch = importFrom.match(renamePackagesRegexp);
+
+    const otterPackage = renamePackageMatch ? renamePackageMatch[0] : importFrom.match(importsRegexp)?.[0];
+
+    // If the import matched an Otter package
+    if (otterPackage) {
+
+      if (!oldImportedSymbolsPerPackage[otterPackage]) {
+        oldImportedSymbolsPerPackage[otterPackage] = [];
+      }
+
+      // We store the import line to be able to remove it afterwards
+      importNodes.push(imp);
+
+      // We retrieve all the symbols listed in the import statement
+      const namedImport = imp.importClause && imp.importClause.getChildAt(0);
+      const imports: ExtractedImport[] = namedImport && ts.isNamedImports(namedImport) ?
+        namedImport.elements.map((element) => ({symbol: element.getText(), location: importFrom})) :
+        [];
+
+      // And associate them to the Otter package
+      oldImportedSymbolsPerPackage[otterPackage].push(...imports);
+    }
+  });
+
+  // If did not capture any interesting import line, do nothing
+  if (importNodes.length === 0) {
+    return 0;
+  }
+
+  // Iterate on the imports we found and lookup into the renamed packages or into the map to see from where the import should be taken
```

Comment:
iterate over

Reply:
Suggestion (I do not know which map you refer to, so maybe you make it more specific)
   // Iterate over the imports and look into the renamed packages and into the map to see the import new module

Reply:
updated